src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1573,6 +1576,430 @@ LedgerManagerImpl::prefetchTransactionData(AbstractLedgerTxnParent& ltx,
+                    txBundle.getEffects().getDelta().header.current =
+                        ltxInner.loadHeader().current();
+                    txBundle.getEffects().getDelta().header.previous =
+                        ltxInner.loadHeader().current();
+                    app.checkOnOperationApply(
+                        txBundle.getTx()->getRawOperations().at(0),
+                        txBundle.getResPayload()->getOpResultAt(0),
+                        txBundle.getEffects().getDelta());
+                }
+                catch (InvariantDoesNotHold& e)
+                {
+                    printErrorAndAbort(
+                        "Invariant failure while applying operations: ",
+                        e.what());
+                }
+            }
+
+            txBundle.getTx()->processPostApply(mApp.getAppConnector(), ltxInner,
+                                               txBundle.getEffects().getMeta(),
+                                               txBundle.getResPayload());
+
+            // We only increase the internal-error metric count if the
+            // ledger is a newer version.
+            if (txBundle.getResPayload()->getResultCode() == txINTERNAL_ERROR &&
+                ledgerInfo.getLedgerVersion() >=
+                    config.LEDGER_PROTOCOL_MIN_VERSION_INTERNAL_ERROR_REPORT)
+            {
+                auto& internalErrorCounter = app.getMetrics().NewCounter(
+                    {"ledger", "transaction", "internal-error"});
+                internalErrorCounter.inc();
+            }
+        }
+    }
+
+    for (auto const& threadEntryMap : entryMapsByThread)
+    {
+        for (auto const& entry : threadEntryMap)
+        {
+            // Only update if dirty bit is set
+            if (!entry.second.isDirty)
+            {
+                continue;
+            }
+
+            if (entry.second.mLedgerEntry)
+            {
+                auto const& updatedEntry = *entry.second.mLedgerEntry;
+                auto ltxe = ltxInner.load(entry.first);
+                if (ltxe)
+                {
+                    if (ltxe.current().data.type() == TTL)
+                    {
+                        auto currLiveUntil =
+                            ltxe.current().data.ttl().liveUntilLedgerSeq;
+
+                        releaseAssertOrThrow(
+                            updatedEntry.data.ttl().liveUntilLedgerSeq >=
+                            currLiveUntil);
+                    }
+                    ltxe.current() = updatedEntry;
+                }
+                else
+                {
+                    ltxInner.create(updatedEntry);
+                }
+            }
+            else
+            {
+                auto ltxe = ltxInner.load(entry.first);
+                if (ltxe)
+                {
+                    ltxInner.erase(entry.first);
+                }
+            }
+        }
+    }
+
+    for (auto const& kvp : cumulativeRoTtlDeltas)
```

Comment:
One thing I haven't realized with the new TTL algorithm is tx meta. We might need to rethink how we emit TTL meta now. In the worst case we'd need to switch back from deltas to the max value, but maybe we can come up with something better...

Reply:
Update the ttl reconciliation logic.